### PR TITLE
feat: updated tags list UI

### PR DESCRIPTION
**Pull request by MehadND**
This PR addresses the issue raised in #3288. 

Modified the UI so that user can easily search tags.

#### Demo (old)

https://github.com/usememos/memos/assets/34424878/d0cef8d9-ceff-4664-b5c8-3610bb5d58a8

### CODE REVIEW

**Comment by MehadND** [comment]
Did not test with subtags (as I didn't know how to make them) so please feel free to test with it.

File: web/src/components/HomeSidebar/TagsSection.tsx [hunk 1]
```diff
@@ -72,121 +70,68 @@ const TagsSection = () => {
...
-  const [showSubTags, toggleSubTags] = useToggle(false);
-
-  const handleTagClick = () => {
-    if (isActive) {
-      filterStore.setTagFilter(undefined);
-    } else {
-      filterStore.setTagFilter(tag.text);
-    }
-  };
+  const [value, setValue] = useState<string | null>();
```
Comment (michaelien) [inline web/src/components/HomeSidebar/TagsSection.tsx:73]: How about `search`?

```suggestion
  const [value, setValue] = useState<string | null>();
```

**Review by michaelien — APPROVED** [review]
LGTM. Could we have some screenshots about the UI part

**Comment by MehadND** [comment]
Few Screenshots of UI:

image

image

**Comment by kbiits** [comment]
Should we care about the subtags?? I think this PR will make the subtags not displayed

**Comment by kbiits** [comment]
@MehadND Just add `/` in your tag to make it a subtag. Ex: `#parent/sub1`. `sub1` is a child of `parent`.
Please look at [screen capture here](https://github.com/usememos/memos/issues/3309)

**Comment by boojack** [comment]
> is there any chance where we can just render the tags and their subtags all together. Like in a long list of tags.

SGTM, updated with https://github.com/usememos/memos/commit/d355e2c631686725ba30c3d4dab92e2ce35f18b1

File: web/src/components/HomeSidebar/TagsSection.tsx [hunk 2]
```diff
@@ -24,6 +24,22 @@ const TagsSection = () => {
     [memoList.size()],
   );
 
+  const [search, setSearch] = useState<string | null>();
+
+  const handleDeleteTag = async (tagValue: string) => {
```
Comment (Dubzer) [inline web/src/components/HomeSidebar/TagsSection.tsx:29]: You can wrap this function in the [useCallback](https://react.dev/reference/react/useCallback) to cache it between re-renders

**Comment by boojack** [comment]
The main branch makes a lot of changes to tags, and there are conflicts in this PR, so let's close it first.

**Pull request closed without merging**